UBC-Thunderbots/Software
# Add referee display layer

### Description
Add a referee info layer to display game state and command.

Testing steps:
1. turn on "Referee Info" option in the "Layers" dropdown
2. Expect to see info on the top right corner


### Testing Done
- Thunderscope runs
- CI



### Resolved Issues
resolves #3333 
resolves #3289



### Demo
<details>
    <summary>Dropdown menu to enable layer</summary>
         <img width="964" alt="image" src="https://github.com/user-attachments/assets/9fb1e93b-79fb-41f6-ba13-4529b4f997fa">
</details>

<details>
    <summary>Game state and command display</summary>
         <img width="513" alt="image" src="https://github.com/user-attachments/assets/996b416a-6681-4d7c-8a18-a763b8b277d0">
</details>


<details>
    <summary>Ball placement visual (ball is not placed correctly)</summary>
         <img width="959" alt="image" src="https://github.com/user-attachments/assets/010b4423-a690-47a1-9921-c298fff9fd27">
</details>

<details>
    <summary>Ball placement visual (ball is placed correctly)</summary>
        <img width="519" alt="image" src="https://github.com/user-attachments/assets/df8b238a-1142-4483-a27b-ebeaea5426e6">
</details>


### Review Checklist



**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
